ingenialink/dictionary.py

```diff
@@ -200,6 +200,8 @@ class Dictionary(ABC):
     """Firmware version declared in the dictionary."""
     product_code: int
     """Product code declared in the dictionary."""
+    coco_product_code: Optional[int] = None
+    """CoCo product code declared in the dictionary."""
```

Comment:
I would add a comment here explaining that it typically comes from comkit dictionary. Otherwise it looks like the dictionary may contain this coco_produc_code.

Other option is to inherit this class and return a instance of a subclass when __add__ is called